src/dali/plugin/input/rest/kraken.py

```diff
@@ -0,0 +1,414 @@
+                return something
+
+        # get initial trade history to get count
+        index: int = 0
+        trade_history = self._client.private_post_tradeshistory(params={_OFFSET: index})
+        count: int = int(trade_history[_RESULT][_COUNT])
+        trade_history = dict()
+        while index < count:
+            trade_history.update(self._process_trade_history(index))
+            index += _TRADE_RECORD_LIMIT
+
+        ledger = self._client.private_post_ledgers(params={_OFFSET: index})
+        index = 0
+        count = int(ledger[_RESULT][_COUNT])
+        ledger = dict()
+        while index < count:
+            ledger.update(self._process_ledger(index))
+            index += _TRADE_RECORD_LIMIT
+
+        result = (trade_history, ledger)
+
+        if self.use_cache:
+            with open(self.__CACHE_FILE, 'wb') as file:
+                pickle.dump(result, file)
+
+        return result
+
+    def load(self) -> List[AbstractTransaction]:
+        (trade_history, ledger) = self._gather_api_data()
+
+        result = {'in': [], 'out': [], 'intra': []}
+
+        unhandled_types = dict()
+        for key, value in ledger.items():
+            if value[_TYPE] == _WITHDRAWAL or value[_TYPE] == _DEPOSIT:
+                result[_INTRA].append(key)
+            elif value[_TYPE] == _TRADE and 'USD' not in value[_ASSET]:
+                if float(value[_AMOUNT]) > 0: result[_IN].append(key)
+                else: result[_OUT].append(key)
+            elif value[_TYPE] == _MARGIN:
+                result[_OUT].append(key)
+            elif value[_TYPE] == _ROLLOVER:
+                result[_OUT].append(key)
+            elif value[_TYPE] == _TRANSFER:
+                result[_IN].append(key)
+            elif value[_TYPE] == _SETTLED:
+                # ignorable in terms of in/out/intra
+                pass
+            elif value[_TYPE] == _CREDIT:
+                # 'credit not implemented'
+                pass
+            elif value[_TYPE] == _STAKING:
+                # 'credit not implemented'
+                pass
+            elif value[_TYPE] == _SALE:
+                # 'credit not implemented'
+                pass
+            else:
+                unhandled_types.update({value[_TYPE]: key})
+        self.__logger.debug(f"unhandled types of the ledger={unhandled_types}")
+
+        return self._compute_tx_set(trade_history, ledger, result)
+
+    def _compute_tx_set(self, trade_history, ledger, processed_transactions) -> List[AbstractTransaction]:
+        result: List[AbstractTransaction] = []
+
+        transactions = processed_transactions[_IN] + processed_transactions[_OUT] + processed_transactions[_INTRA]
+        for key in transactions:
+            record = ledger[key]
+            timestamp_value: datetime = datetime.fromtimestamp(float(record[_TIMESTAMP]))
+            timestamp_value = timestamp_value.astimezone(self.__timezone)
+            timestamp_value = self.__timezone.normalize(timestamp_value)
```

Comment:
You can use `_rp2_timestamp_from_ms_epoch` to convert millisecond epoch to a RP2 timestamp:
```
timestamp_value: str = self._rp2_timestamp_from_ms_epoch(record[_TIMESTAMP])
```